sdk/python/feast/offline_store.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2019 The Feast Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from abc import ABC, abstractmethod
+from datetime import datetime
+from typing import Dict, List, Optional
+
+import pyarrow
+
+from feast.entity import Entity
+from feast.feature import Feature
+from feast.feature_view import FeatureView
+
+
+class OfflineStore(ABC):
+    """
+    OfflineStore is a non-user-facing object used for all interaction between Feast and the service used for offline storage of features. Currently BigQuery is supported.
+    """
+
+    @abstractmethod
+    def pull_table(
+        self, feature_view: FeatureView, start_date: datetime, end_date: datetime,
+    ):
+        pass
+
+
+class BigQueryOfflineStore:
+    """
+    BigQueryOfflineStore is a non-user-facing object used for all interaction between Feast and BigQuery.
+    """
+
+    def pull_table(
+        self, feature_view: FeatureView, start_date: datetime, end_date: datetime,
+    ) -> Optional[pyarrow.Table]:
+        from google.cloud.bigquery_storage import BigQueryReadClient, types
+
+        project, dataset, table = feature_view.inputs.table_ref.split(".")
+        client = BigQueryReadClient()
+
+        requested_session = types.ReadSession()
+        requested_session.table = (
+            f"projects/{project}/datasets/{dataset}/tables/{table}"
+        )
+        requested_session.data_format = types.DataFormat.ARROW
+
+        # if we have mapped fields, use the original field names in the call to BigQuery
+        event_timestamp_column = feature_view.inputs.event_timestamp_column
+        fields = (
```

Comment:
Can we please add a few comments to this code so that its easier for a reader to parse